pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -490,6 +495,72 @@ func generateNetworkEnhancedMetrics(args generateNetworkEnhancedMetricArgs) {
 	})
 }
 
+type generateTmpEnhancedMetricsArgs struct {
+	TmpMax  float64
+	TmpUsed float64
+	Tags    []string
+	Demux   aggregator.Demultiplexer
+	Time    float64
+}
+
+// generateTmpEnhancedMetrics generates enhanced metrics for space used and available in the /tmp directory
+func generateTmpEnhancedMetrics(args generateTmpEnhancedMetricsArgs) {
+	args.Demux.AggregateSample(metrics.MetricSample{
+		Name:       tmpUsedMetric,
+		Value:      args.TmpUsed,
+		Mtype:      metrics.DistributionType,
+		Tags:       args.Tags,
+		SampleRate: 1,
+		Timestamp:  args.Time,
+	})
+	args.Demux.AggregateSample(metrics.MetricSample{
+		Name:       tmpMaxMetric,
+		Value:      args.TmpMax,
+		Mtype:      metrics.DistributionType,
+		Tags:       args.Tags,
+		SampleRate: 1,
+		Timestamp:  args.Time,
+	})
+}
+
+func SendTmpEnhancedMetrics(sendMetrics chan bool, tags []string, metricAgent *ServerlessMetricAgent) {
+	if enhancedMetricsDisabled {
+		return
+	}
+
+	bsize, blocks, bavail, err := statfs(tmpPath)
+	if err != nil {
+		log.Debug("Could not emit tmp enhanced metrics")
+		return
+	}
+	tmpMax := blocks * bsize
+	tmpUsed := bsize * (blocks - bavail)
+
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-sendMetrics:
```

Comment:
This is the only spot where we're consuming from this channel. In the case where this function returns early (and therefore stops consuming) then any writes to the channel will block forever.

Reply:
I refactored to stop the ticker and keep the loop going instead of returning, but I'm open to other ideas if there's a better way of handling this